checker/manual/constant-value-checker.tex

```diff
@@ -38,22 +38,31 @@ \subsection{Type Annotations\label{constant-value-checker-type-annotations}}
 The type annotations are
 \refqualclass{common/value/qual}{BoolVal},
 \refqualclass{common/value/qual}{IntVal},
+\refqualclass{common/value/qual}{IntRange},
 \refqualclass{common/value/qual}{DoubleVal}, and
 \refqualclass{common/value/qual}{StringVal}.
 
-Each type annotation takes as an argument a set of values, and its meaning
+Each \<*Val> type annotation takes as an argument a set of values, and its
+meaning
 is that at run time, the expression evaluates to one of the values.  For
 example, an expression of type
 \<\refqualclass{common/value/qual}{StringVal}("a", "b")> evaluates to
 one of the values \<"a">, \<"b">, or \<null>.
 The set is limited to 10 entries; if a variable
 could be more than 10 different values, the Constant Value
 Checker gives up and its type becomes
-\refqualclass{common/value/qual}{UnknownVal} instead.
+\refqualclass{common/value/qual}{IntRange} for integral types and
+\refqualclass{common/value/qual}{UnknownVal} for all other types.
 
 % \refqualclass{checker/value/qual}{BottomVal}, meaning that the expression
 % is dead or always has the value \<null>.
 
+\refqualclass{common/value/qual}{IntRange} takes two arguments --- a lower
```

Comment:
The type hierarchy figure does not include @IntRange, but it needs to.  The manual should explain the subtyping relationship between @IntRange and @IntVal (and there should be tests that verify that the implementation acts properly).